src/Domain/Handler/SubscriptionPurchasedEventHandler.ts

```diff
@@ -0,0 +1,96 @@
+import { RoleName, SubscriptionName } from '@standardnotes/auth'
+import {
+  DomainEventHandlerInterface,
+  SubscriptionPurchasedEvent,
+  SubscriptionRenewedEvent,
+} from '@standardnotes/domain-events'
+import { inject, injectable } from 'inversify'
+import { Logger } from 'winston'
+
+import TYPES from '../../Bootstrap/Types'
+import { RoleRepositoryInterface } from '../Role/RoleRepositoryInterface'
+import { User } from '../User/User'
+import { UserRepositoryInterface } from '../User/UserRepositoryInterface'
+import { UserSubscription } from '../User/UserSubscription'
+import { UserSubscriptionRepositoryInterface } from '../User/UserSubscriptionRepositoryInterface'
+
+@injectable()
+export class SubscriptionPurchasedEventHandler
+implements DomainEventHandlerInterface
+{
+  constructor(
+    @inject(TYPES.UserRepository) private userRepository: UserRepositoryInterface,
+    @inject(TYPES.RoleRepository) private roleRepository: RoleRepositoryInterface,
+    @inject(TYPES.UserSubscriptionRepository) private userSubscriptionRepository: UserSubscriptionRepositoryInterface,
+    @inject(TYPES.Logger) private logger: Logger
+  ) {}
+
+  private subscriptionNameToRoleNameMap = new Map<SubscriptionName, RoleName>([
```

Comment:
could you put private functions on the bottom of the class? So that when opening the file you have the public ones on first eye sight